cl/disclosures/management/commands/import_financial_disclosures.py

```diff
@@ -304,165 +27,39 @@ def has_been_extracted(data: Dict[str, Union[str, int, list]]) -> bool:
-    """Get URL saved to download filepath
-
-    :param data: File data
-    :return: URL or first URL on AWS
-    """
-    if data["disclosure_type"] == "jw" or data["disclosure_type"] == "single":
-        url = data["url"]
-    else:
-        url = data["urls"][0]
-    return url
-
-
-def get_disclosure_from_pdf_path(disclosure_url: str) -> FinancialDisclosure:
-    """Convenience method to get disclosure from download filepath
-
-    :param disclosure_url: The URL of the first link (if there are more than
-    one) of the source FD tiff(s)/PDF
-    :return: Financial Disclosure object
-    """
-    return FinancialDisclosure.objects.get(download_filepath=disclosure_url)
-
-
-def has_been_pdfed(disclosure_url: str) -> Optional[str]:
-    """Has file been PDFd from tiff and saved to AWS.
-
-    :param disclosure_url: The URL of the first link (if there are more than
-    one) of the source FD tiff(s)/PDF
-    :return: Path to document or None
-    """
-
-    disclosures = FinancialDisclosure.objects.filter(
-        download_filepath=disclosure_url
-    )
-    if disclosures.exists():
-        return (
-            f"https://{settings.AWS_S3_CUSTOM_DOMAIN}/"
-            f"{disclosures[0].filepath}"
-        )
-
-
-def generate_or_download_disclosure_as_pdf(
-    data: Dict[str, Union[str, int, List[str]]],
-    pdf_url: Optional[str],
-) -> requests.Response:
-    """Generate or download PDF content from images or urls.
-
-    :param data: Data to process.
-    :param pdf_url: The URL of PDF in S3
-    :return: Response containing PDF
-    """
-    if pdf_url:
-        logger.info(f"Downloading PDF: {pdf_url}")
-        return requests.get(pdf_url, timeout=60 * 20)
-    elif data["disclosure_type"] == "jw":
-        logger.info(f"Downloading JW PDF: {quote(data['url'], safe=':/')}")
-        return requests.get(data["url"], timeout=60 * 20)
-    elif data["disclosure_type"] == "single":
-        urls = [data["url"]]
-    else:
-        urls = data["urls"]
-    logger.info(f"Processing url:{quote(urls[0], safe=':/')}")
-    return requests.post(
-        settings.BTE_URLS["images-to-pdf"]["url"],
-        json=json.dumps({"urls": urls}),
-        timeout=settings.BTE_URLS["images-to-pdf"]["timeout"],
-    )
-
-
 def import_financial_disclosures(
-    filepath: str,
-    skip_until: Optional[str],
-):
-    """Import financial documents into courtlistener
+    filepath: str, skip_until: Optional[str], queue_name: str, min_size: int
+) -> None:
+    """Import financial documents into courtlistener.
 
     :param filepath: Path to file data to import.
     :param skip_until: ID if any to skip until.
+    :param queue_name: The celery queue name
+    :param min_size: The minimum items in a queue
     :return:None
     """
+    throttle = CeleryThrottle(
+        queue_name=queue_name,
+        min_items=min_size,
+    )
     with open(filepath) as f:
         disclosures = json.load(f)
 
     for data in disclosures:
+        throttle.maybe_wait()
```

Comment:
The throttle monitors the speed that the queue is being consumed and adjusts how long to wait accordingly. As a result, if you put it before your skipping logic, I believe it will think things are going *really* fast, and it'll queue up more than it should. Instead, I believe, the maybe_wait() command should go below your skipping logic, so it's only called when needed.